server/query_test.go

```diff
@@ -124,10 +116,18 @@ func TestParseQuery(t *testing.T) {
 			`file:HELLO`,
 			pb.Query{Line: "HELLO", FoldCase: false, FilenameOnly: true},
 		},
+		{
+			`lit:a( file:b`,
+			pb.Query{Line: `a\(`, File: "b", FoldCase: false},
+		},
+		{
+			`lit:a(b file:c`,
+			pb.Query{Line: `a\(b`, File: "c", FoldCase: false},
+		},
 	}
 
 	for _, tc := range cases {
-		parsed, err := ParseQuery(tc.in)
+		parsed, err := ParseQuery(tc.in, true)
```

Comment:
Would it make sense to roll the "is regex" flag into the test table, instead of having two loops? That would certainly be my instinct.

If you wanted to be slightly more concise, we could even flip the polarity and have the flag be "isLiteral" so that you could omit it from the regex tests and it'd default to false.

Reply:
I think I tried that and instead of defaulting to false I just got an error that I wasn't providing enough fields to initialize. Maybe I'm misremembering, so I'll give it another go.

Reply:
Mm, that might require switching to explicit field names, which may not be a great tradeoff.

I'd weakly prefer spelling out `false` and `true` in each case over two loops, I think.